src/commitment.rs

```diff
@@ -56,6 +65,36 @@ pub fn create<'a>(polynomial: &'a polynomial::Polynomial, setup: &setup::Setup)
     }
 }
 
+impl Opening {
+    pub fn verify(&self, setup: &setup::Setup) -> bool {
+        unsafe {
+            let g1 = blst::blst_p1_generator();
+            let g2 = blst::blst_p2_generator();
+
+            // Compute LHS
+            // 1. [f(s) - y]_1
+            let mut y_scalar = blst::blst_scalar::default();
+            blst::blst_scalar_from_fr(&mut y_scalar, &self.value);
+
+            let mut neg_y_in_g1 = blst::blst_p1::default();
+            blst::blst_p1_mult(&mut neg_y_in_g1, g1, &y_scalar, constants::MODULUS_BIT_SIZE);
+            blst::blst_p1_cneg(&mut neg_y_in_g1, true); // FIXME(sproul): wtf does the bool do
```

Comment:
i'm pretty sure that the flag is referring to one of these bits here: https://tools.ietf.org/html/draft-irtf-cfrg-pairing-friendly-curves-07#appendix-C.1

and in fact it may even be the `C_bit`... i would write a little test just to verify that this being `true` or `false` matches your intuition -- check out `point::negate` for an example (although in blst it seems that the high-order bit is used to indicate sign in `Fr`).

Reply:
I think it might be a constant-time flag, but I'm not sure. Ben Edginton seems to use `true` consistently in `c-kzg`:

https://github.com/benjaminion/c-kzg/blob/66c50a49b1811d743fba195dca042d262674c2b4/src/bls12_381.c#L367

But I shall investigate further.

Reply:
cneg is for conditional negation. If you want to always negate you need to pass true, if you pass false it will leave the input untouched.

Reply:
awesome, thanks @mratsim!